README.md

```diff
@@ -1,2 +1,17 @@
 # go-http-middleware
 
+[![Build Status](https://drone.corp.mongodb.com/api/badges/kanopy-platform/go-http-middleware/status.svg)](https://drone.corp.mongodb.com/kanopy-platform/go-http-middleware)
+
+The `go-http-middleware` is a collection of generic HTTP middleware functions.
+
+## Design Guidelines
+
+* Middleware MUST NOT rely on external application specific state
```

Comment:
Not entirely sure what this means. Does this mean this shouldn't assume that certain state exists or that any dependencies should be passed as a middleware func argument, etc.?

Reply:
I'll adjust the wording here to focus on loose coupling design principles.